File: chess_cpp/validators/EmptySlotValidator.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "EmptySlotValidator.h"

bool EmptySlotValidator::validate(Board *board, Spot *start, Spot *end) {
    return start != nullptr && start->getPiece() != nullptr;
}

std::string EmptySlotValidator::getNotValidMessage() {
    return MessageHandler::getHellNoMessage();
}

File: chess_java/src/chess/validators/EmptySlotValidator.java
package chess.validators;

import chess.Board;
import chess.Spot;
import chess.utils.MessageHandler;

public class EmptySlotValidator extends PieceMoveValidator {

    @Override
    public boolean validate(Board board, Spot start, Spot end) {
        return start != null && start.getPiece() != null;
    }

    @Override
    protected String getNotValidMessage() {
        return MessageHandler.getHellNoMessage();
    }
}
